### PR TITLE
Enable cursor support for PostgreSQL and SQLite

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

This PR completely removes the testing option for enabling and disabling cursors; they are always enabled now.

For everyone to review, hence https://github.com/FerretDB/FerretDB/labels/do%20not%20merge.

## Readiness checklist

- [ ] I added/updated unit tests.
- [ ] I added/updated integration/compatibility tests.
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/backends/sqlite/query_iterator.go
Comment (w84thesun) [inline internal/backends/sqlite/query_iterator.go:1]: Is it possible to extract those changes into a separate PR? They would be handy in the `delete` issue I'm working on.

File: internal/handlers/pg/msg_find.go [hunk 1]
```diff
@@ -50,11 +50,13 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 		return nil, err
 	}
 
-	if params.MaxTimeMS != 0 {
-		ctxWithTimeout, cancel := context.WithTimeout(ctx, time.Duration(params.MaxTimeMS)*time.Millisecond)
-		defer cancel()
+	username, _ := conninfo.Get(ctx).Auth()
 
-		ctx = ctxWithTimeout
+	cancel := func() {}
```
Comment (chilagrow) [inline internal/handlers/pg/msg_find.go:55]: I'm thinking that it's easy to forget calling `cancel()` on newly added error, and leave resources behind. Maybe this is not a concern as we will cleanup unused cursors.

Alternative could be always cancelling cursor upon any error something like this, but it's not ideal as `cancel()` shouldn't be cancelled if cursor is created.
```
defer func(err error){
if err !=nil{
cancel()
}
}(err)
```

**Review by chilagrow — APPROVED** [review]
You are right about the complexity added by cursor.

Overall looks great. We have to watch out on error handling when cursor is involved.

File: internal/handlers/pg/msg_find.go (hunk 1 above)
Comment (AlekSi) [reply internal/handlers/pg/msg_find.go:55]: `go vet` catches those problems:

```
-> task lint
task: [lint] bin/golangci-lint run --config=.golangci.yml
internal/handlers/pg/msg_find.go:59:3: lostcancel: the cancel function is not used on all paths (possible context leak) (govet)
                ctx, cancel = context.WithTimeout(ctx, time.Duration(params.MaxTimeMS)*time.Millisecond)
                ^
internal/handlers/pg/msg_find.go:72:4: lostcancel: this return statement may be reached without using the cancel var defined on line 59 (govet)
                        return nil, err
                        ^
task: Failed to run task "lint": exit status 1
```

File: internal/handlers/pg/msg_aggregate.go [hunk 2]
```diff
@@ -200,31 +203,33 @@ func (h *Handler) MsgAggregate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
...
+	})
 
-	if h.EnableCursors && int64(len(resDocs)) > batchSize {
-		// Cursor is not created when resDocs is less than batchSize, it all fits in the firstBatch.
-		iter := iterator.Values(iterator.ForSlice(resDocs))
-		c := cursor.New(&cursor.NewParams{
-			Iter:       iter,
-			DB:         db,
-			Collection: collection,
-			BatchSize:  int32(batchSize),
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply internal/handlers/pg/msg_aggregate.go:212]: That's an open question. I think we should have tests for all four combinations of `batchSize` being present/absent in `find` and `getMore`. Do we have them? If not, could you add them in your PR?

File: internal/handlers/pg/msg_find.go (hunk 1 above)
Comment (chilagrow) [reply internal/handlers/pg/msg_find.go:55]: Good to know, yay for linting :+1:

File: internal/handlers/pg/msg_aggregate.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chilagrow) [reply internal/handlers/pg/msg_aggregate.go:212]: Yeah I will add them.

File: internal/clientconn/cursor/cursor.go [hunk 3]
```diff
@@ -16,38 +16,81 @@
...
+// newCursor creates a new cursor.
+func newCursor(id int64, db, collection, username string, iter types.DocumentsIterator, r *Registry) *Cursor {
+	c := &Cursor{
+		ID:         id,
+		DB:         db,
+		Collection: collection,
+		Username:   username,
+		iter:       iter,
+		r:          r,
+		created:    time.Now(),
```
[the remark this answers is not in the document]
Comment (chilagrow) [reply internal/clientconn/cursor/cursor.go:64]: Oh my comment isn't obvious, I meant in the future we may store datetime for when Next() is called too

[the remark this answers is not in the document]
Comment (AlekSi) [reply internal/clientconn/cursor/cursor.go:64]: Maybe tracking intervals between `getMore` or `Next()` calls could be useful, but I'm not sure what Prometheus metric type to use there

[the remark this answers is not in the document]
Comment (chilagrow) [reply internal/clientconn/cursor/cursor.go:64]: Yeah I'm not sure either, I would have to look
